common/persistence/persistence-tests/historyV2PersistenceTest.go

```diff
@@ -174,6 +181,82 @@ func (s *HistoryV2PersistenceSuite) TestScanAllTrees() {
 	s.Equal(0, len(trees))
 }
 
+func (s *HistoryV2PersistenceSuite) TestHistoryScavenger() {
+	controller := gomock.NewController(s.T())
+	historyServiceClient := historyservicemock.NewMockHistoryServiceClient(controller)
+	scavenger := history.NewScavenger(
```

Comment:
Persistence tests should focus on testing the persistence api I think and should not test scavenger impl.

If e2e integration test is needed, those tests should live under `host`.